src/lib/src/check_cpu_speed.c

```diff
@@ -0,0 +1,65 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+unsigned long fibonacci(unsigned long n) {
+    if (n == 0 || n == 1) {
+        return n;
+    }
+    return fibonacci(n - 1) + fibonacci(n - 2);
+}
+
+void benchmark_fibonacci(int iterations) {
+    printf("Running Test 1...\n");
+
+    // Calculate fibonacci (single core check)
+    double total_time = 0.0;
+    for (int i = 0; i < iterations; i++) {
+        clock_t start = clock();
+        fibonacci(45);
+        clock_t end = clock();
+        total_time += ((double)(end - start)) / CLOCKS_PER_SEC;
+    }
+    double average_time = total_time / iterations;
+
+    printf("Test 1: average time taken: %f seconds\n", average_time);
+}
+
+void benchmark_gzip(int iterations) {
+    printf("\nRunning Test 2...\n");
+
+    double total_time = 0.0;
+    for (int i = 0; i < iterations; i++) {
+
+        // since a child process is created using the clock() won't be accurate anymore as clock() measures the CPU
+        // spent time by the program it self, so any actual spent time from the child process will not be taken into account.
+        // Thus we switch to using the actual wall time instead for this benchmark.
+        struct timespec start, end;
+        clock_gettime(CLOCK_MONOTONIC_RAW, &start);
+
+        int status = system("head -1000000 /dev/urandom | gzip > /dev/null");
```

Comment:
you need to do `head -100000b` to specify bytes. without the `b` it reads 1000000 lines which is not a fixed amount of data (depends on where newline characters happen to appear).

Reply:
Thanks @mystenmark for the comment. When I tried with the `b` in the end it wasn't working for me
```
head: illegal line count -- 100000b
``` 
and the examples that I saw on the internet they were stating that using the `head -number` would work equally - but now that I am looking this further it seems wrong.

By the head manual it seems that I need to use the `-c` option to count bytes. Indeed when counting the produced bytes I get:

```
$ head  -c 1000000 /dev/urandom | wc -c | awk '{ printf "%.2f MB\n", $1/1024/1024 }'
0.95 MB
```

whereas with
```
$ head  -1000000 /dev/urandom | wc -c | awk '{ printf "%.2f MB\n", $1/1024/1024 }' 
244.19 MB
```

which as you said is simply wrong for fulfilling the command expectation.

I'll update the command to use:
```
head  -c 268435456 /dev/urandom | gzip > /dev/null
```

to test compressing `256 MB` worth of data